src/main/java/org/mule/extension/socket/internal/SocketUtils.java

```diff
@@ -174,17 +175,28 @@ public static void sendUdpPackages(InputStream inputStream, int bufferSize, Sock
   }
 
   /**
-   * Logs the content if it's log enabled, returns the same content.
+   * Series of static methods that log the content if it's log-debug level enabled,
+   * returning either void or the same content.
    */
-  public static InputStream logIfNeeded(InputStream content, Logger logger) {
+  public static InputStream logIfDebugEnabled(InputStream content, Logger logger) {
     if (logger.isDebugEnabled()) {
-      String c = IOUtils.toString(content);
-      logger.debug("Logging TCP Content (Base64 encoding):");
-      logger.debug("-----------------------------------");
-      logger.debug(Base64.getEncoder().encodeToString(c.getBytes()));
-      logger.debug("-----------------------------------");
-      return new ByteArrayInputStream(c.getBytes());
+      byte[] bytes = IOUtils.toByteArray(content);
```

Comment:
I don't recommend doing this. You might be loading a lot of data into memory. Read and Log chunks of the input stream. There's some challenge having to return an InputStream.